sycl/include/sycl/multi_ptr.hpp

```diff
@@ -386,11 +382,8 @@ class __SYCL_TYPE(multi_ptr) multi_ptr {
            Space == access::address_space::ext_intel_global_host_space)>>
   explicit
   operator multi_ptr<ElementType, GlobalSpace, DecorateAddress>() const {
-    using global_pointer_t =
-        typename multi_ptr<ElementType, GlobalSpace,
-                           access::decorated::yes>::pointer;
     return multi_ptr<ElementType, GlobalSpace, DecorateAddress>(
-        detail::cast_AS<global_pointer_t>(get_decorated()));
+        detail::static_address_cast<GlobalSpace>(get_decorated()));
```

Comment:
These extension address spaces *are* subsections of the `global_space`, so the cast is known to succeed.